tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1235,7 +1235,7 @@ Status StorageManager::get_fragment_info(
       *array.encryption_key(),
       &array_schema,
       &fragment_metadata,
-      timestamp_start,
+      0,
```

Comment:
Another idea here @KiterLuc is to differentiate between dense and sparse arrays. The anterior ndrange is relevant only to dense arrays in consolidation, so the sparse arrays do not need to get the penalty here.

Reply:
> Another idea here @KiterLuc is to differentiate between dense and sparse arrays. The anterior ndrange is relevant only to dense arrays in consolidation, so the sparse arrays do not need to get the penalty here.

done!

Reply:
> Hmm I am not sure about this change. Note that this function is used by the C APIs as well, not just the consolidator. I think we need to duplicate this function after all - one for the consolidator and one for the fragment info C API.
> 
> cc-ing @nguyenv who is working on the C APIs. I think it is ok to accept this change here, but @nguyenv will have to overload this function and instead reopen the array with the proper timestamp range and also calling `reopen` from the input `Array` object (which now needs to be passed by pointer). We had an internal discussion about that.

@stavrospapadopoulos as far as I can see (maybe I'm wrong) the only call from the c_api to this opens with timestamp 0 so the additional code for anterior range will not be run.

Reply:
We have an issue in a separate line of work, which requires changing this function to pass `Array` by pointer and have `reopen` happen in the `Array` object, which in this case it would bring unnecessary fragments that would unnecessarily impact performance. I will work with @nguyenv separately to address this by writing a separate function.